src/Agent.Listener/Configuration/WindowsServiceControlManager.cs

```diff
@@ -26,13 +26,13 @@ public override void Initialize(IHostContext hostContext)
             _term = HostContext.GetService<ITerminal>();
         }
 
-        public void ConfigureService(AgentSettings settings, CommandSettings command)
+        public void ConfigureService(AgentSettings settings, CommandSettings command, string agentType)
```

Comment:
From separation of concern, SCM having to understand agenttype seem like a wrong one. Can we move the responsibility to check the agent type to layer above.